src/background/responders.ts

```diff
@@ -103,14 +107,16 @@ export const responders: { [T in Action['type']]: Responder<T> } = {
         },
       },
     })
+
     return { tabs: nextTabs }
   },
   FETCH_HANDLE_SUCCESS(state, action): Partial<AppState> {
-    const { handle } = action.payload
-    const tab = activeTab(state)
-
-    // If the tab doesn't exist anymore, don't try to update it
+    const { tabId, host, handle } = action.payload
+    const tab = state.tabs.get(tabId)
+    // If the tab doesn't exist anymore, or if the host has since changed, don't try to update it
     if (!tab) return {}
+    if (tab.host !== host) return {}
```

Comment:
I agree that checking the host is important because fetching the handle is an async operation, but so is taking a screenshot and posting a tweet. Shouldn't we apply the same check for the other `SUCCESS`/`FAILURE` actions?